File: final-prices-with-a-special-discount-in-a-shop/final-prices-with-a-special-discount-in-a-shop.cpp
class Solution {
public:
    vector<int> finalPrices(vector<int>& prices) {
        vector<int>res;
        int n = prices.size();
        for(int i = 0 ; i < n ; i++){
            for(int j = i+1 ; j < n ; j++){
                if(prices[i]>=prices[j]){
                    prices[i]= prices[i]-prices[j] ;
                    break;
                }

            }
        res.push_back(prices[i]);
        }
        return res;
    }
};

File: final-prices-with-a-special-discount-in-a-shop/final-prices-with-a-special-discount-in-a-shop.py
class Solution:
    def finalPrices(self, prices: List[int]) -> List[int]:
        res = []
        for i in range(len(prices)):
            for j in range(i+1, len(prices)):
                if(prices[i]>=prices[j]):
                    prices[i]=prices[i]-prices[j]
                    break
            res.append(prices[i])
        return res
            